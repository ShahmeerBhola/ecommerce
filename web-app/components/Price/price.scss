.on-sale-price {
  .regular-price {
    position: relative;
    text-decoration: none;

    &::before {
      border-top: 2px solid rgba(242, 57, 57, 0.7);
      bottom: 0;
      content: ' ';
      display: block;
      height: 12px;
      left: 0;
      position: absolute;
      transform: rotate(-7deg);
      width: 100%;
    }
  }
}
