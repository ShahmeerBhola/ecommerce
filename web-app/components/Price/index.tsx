import React from 'react';
import Text from '../Text';
import { formatPrice as _formatPrice, formatVariationsPrice, productHasPriceRange } from '../../utils';

import './price.scss';

export type Props = {
    size?: Styles.FontSizes;
    weight?: Styles.FontWeights;
    price: number;
    salePrice?: number;
    variations?: WordPress.ProductVariation[];
    hideVariationPricing?: boolean;
    showDecimal?: boolean;
};

const Price: React.FunctionComponent<Props> = ({
    price,
    salePrice,
    variations,
    hideVariationPricing = false,
    weight = 'font-medium',
    size = 'text-base',
    showDecimal,
}) => {
    const formatPrice = (_price: number): string => _formatPrice(_price, showDecimal);

    const determinePriceString = (): string => {
        if (!hideVariationPricing && variations && productHasPriceRange(variations)) {
            return formatVariationsPrice(variations);
        }
        return formatPrice(price);
    };

    const NormalPrice = (): JSX.Element => (
        <Text color="text-red-100" weight={weight} size={size} className="product-price">
            {determinePriceString()}
        </Text>
    );

    const SalePrice = (): JSX.Element => (
        <div className="on-sale-price items-center">
            <Text color="text-gray-100" weight={weight} size="text-sm" className="regular-price inline-block">
                {formatPrice(price)}
            </Text>
            <Text color="text-red-100" weight={weight} size={size} className="sale-price">
                {formatPrice(salePrice as number)}
            </Text>
        </div>
    );

    if (salePrice && salePrice !== price) {
        return <SalePrice />;
    }
    return <NormalPrice />;
};

export default Price;
