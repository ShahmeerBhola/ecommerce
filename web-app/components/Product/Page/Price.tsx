import React, { FC, HTMLAttributes, useContext } from 'react';
import isEmpty from 'lodash/isEmpty';

import Price from '../../Price';

import { ProductContext } from './Context';

const ProductPrice: FC<HTMLAttributes<HTMLDivElement>> = (props) => {
    const {
        state: { product, variation },
    } = useContext(ProductContext);

    const { price, sale_price, variations } = product as WordPress.Product;
    const { price: variationPrice, sale_price: variationSalePrice } = variation || {};

    return (
        <div {...props}>
            <Price
                price={variationPrice || price}
                salePrice={variationSalePrice || sale_price}
                variations={variations}
                hideVariationPricing={!isEmpty(variation)}
                weight="font-bold"
                size="text-xl"
            />
        </div>
    );
};

export default ProductPrice;
