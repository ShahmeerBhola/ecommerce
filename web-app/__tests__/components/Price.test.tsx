import React from 'react';
import renderer from 'react-test-renderer';
import Price from '../../components/Price';

describe('Price component', () => {
  test('it renders properly', () => {
    const tree = renderer.create(<Price price={24.99} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('it renders properly with size prop', () => {
    const tree = renderer.create(<Price price={28.49} size="2xl" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('it renders properly with weight prop', () => {
    const tree = renderer.create(<Price price={28.49} weight="bold" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
