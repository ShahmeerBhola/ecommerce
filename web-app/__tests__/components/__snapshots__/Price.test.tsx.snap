// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Price component it renders properly 1`] = `
<p
  className="font-secondary font-medium text-red-100 text-base"
>
  $24.99
</p>
`;

exports[`Price component it renders properly with size prop 1`] = `
<p
  className="font-secondary font-medium text-red-100 text-2xl"
>
  $28.49
</p>
`;

exports[`Price component it renders properly with weight prop 1`] = `
<p
  className="font-secondary font-bold text-red-100 text-base"
>
  $28.49
</p>
`;
